RRS/accounts/receivers.py

```diff
@@ -0,0 +1,32 @@
+from allauth.account import signals as allauth_signals
+from django.dispatch import receiver
+
+PROVIDER_FACEBOOK = 'facebook'
+PROVIDER_GOOGLE = 'google'
+PROVIDER_TWITTER = 'twitter'
+FACEBOOK_FIRST_NAME_DATA = 'first_name'
+FACEBOOK_LAST_NAME_DATA = 'last_name'
+GOOGLE_FIRST_NAME_DATA = 'given_name'
+GOOGLE_LAST_NAME_DATA = 'family_name'
+FULL_NAME_DATA = 'name'
+
+
+@receiver(allauth_signals.user_signed_up)
+def user_signed_up(request, user, social_login=None, **kwargs):
+    if social_login:
+        user.first_name, user.last_name = get_name_from_social_account(
+            social_login.account)
+        user.save()
+
+
+def get_name_from_social_account(account):
+    '''Returns first_name, last_name data of account'''
+    if account.provider == PROVIDER_FACEBOOK:
+        return account.extra_data[FACEBOOK_FIRST_NAME_DATA], account.extra_data[FACEBOOK_LAST_NAME_DATA]
+
+    elif account.provider == PROVIDER_GOOGLE:
+        return account.extra_data[GOOGLE_FIRST_NAME_DATA], account.extra_data[GOOGLE_LAST_NAME_DATA]
+
+    elif account.provider == PROVIDER_TWITTER:
+        name = account.extra_data[FULL_NAME_DATA]
+        return name.split()[0], name.split()[1]
```

Comment:
You can remove the tuple()

On Mon, Apr 20, 2015 at 11:00 AM, Noel Sison II <notifications@github.com>
wrote:

> In RRS/accounts/receivers.py
> <https://github.com/KRGG/RRS/pull/39#discussion_r28695991>:
>
> > +        user.first_name, user.last_name = get_name_from_social_account(
> > +            social_login.account)
> > +        user.save()
> > +
> > +
> > +def get_name_from_social_account(account):
> > +    '''Returns first_name, last_name data of account'''
> > +    if account.provider == PROVIDER_FACEBOOK:
> > +        return account.extra_data[FACEBOOK_FIRST_NAME_DATA], account.extra_data[FACEBOOK_LAST_NAME_DATA]
> > +
> > +    elif account.provider == PROVIDER_GOOGLE:
> > +        return account.extra_data[GOOGLE_FIRST_NAME_DATA], account.extra_data[GOOGLE_LAST_NAME_DATA]
> > +
> > +    elif account.provider == PROVIDER_TWITTER:
> > +        name = account.extra_data[FULL_NAME_DATA]
> > +        return name.split()[0], name.split()[1]
>
> (name.split()[:2] returns a list. Does the conversion to tuple have an
> effect on efficiency?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/KRGG/RRS/pull/39/files#r28695991>.
>